.github/workflows/deployment.yml

```diff
@@ -0,0 +1,92 @@
+#
+# This source file is part of the Stanford Biodesign Digital Health ENGAGE-HF open-source project
+#
+# SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+#
+# SPDX-License-Identifier: MIT
+#
+
+name: Deployment
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: |
+          The GitHub deployment environment.
+        required: true
+        default: 'development'
+        type: choice
+        options:
+          - development
+          - staging
+          - production
+  workflow_call:
+    inputs:
+      environment:
+        description: |
+          The GitHub deployment environment.
+        required: false
+        type: string
+        default: development
+
+concurrency:
+  group: deployment
+  cancel-in-progress: false
+
+jobs:
+  determineenvironment:
+    name: Determine Environment
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.set-env.outputs.environment }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Determine Environment
+        id: set-env
+        run: |
+          if [[ -z "${{ inputs.environment }}" ]]; then
+            echo "environment=staging" >> $GITHUB_OUTPUT
+          else
+            echo "environment=${{ inputs.environment }}" >> $GITHUB_OUTPUT
+          fi
+  vars:
+    name: Inject Environment Variables In Deployment Workflow
+    needs: determineenvironment
+    runs-on: ubuntu-latest
+    environment: ${{ needs.determineenvironment.outputs.environment }}
+    outputs:
+      FIREBASE_PROJECT_ID: ${{ vars.FIREBASE_PROJECT_ID }}
+    steps:
+      - run: |
+          echo "Injecting Environment Variables In Deployment Workflow: ${{ vars.FIREBASE_PROJECT_ID }}"
+  buildandtest:
+    name: Build and Test
+    needs: determineenvironment
+    uses: ./.github/workflows/build-and-test.yml
+    permissions:
+      contents: read
+      checks: write
+    secrets: inherit
+  deployfirebase:
+    name: Deploy Firebase Project
+    needs: [buildandtest, determineenvironment, vars]
+    uses: StanfordBDHG/.github/.github/workflows/firebase-deploy.yml@envFirebase
+    permissions:
+      contents: read
+    with:
+      environment: ${{ needs.determineenvironment.outputs.environment }}
+      arguments: '--project ${{ needs.vars.outputs.FIREBASE_PROJECT_ID }}'
+    secrets:
+      GOOGLE_APPLICATION_CREDENTIALS_BASE64: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS_BASE64 }}
+      ENV_FILE: |
```

Comment:
I think this can be removed. `ENV_FILE` is used only for `docker-compose-test.yml` action, it's not used in `firebase-deploy.yml`

Reply:
[Remove ENV_FILE](https://github.com/StanfordBDHG/ENGAGE-HF-Web-Frontend/pull/40/commits/0bf241106288bfd88c1fcbc7c300b7b7fcb1b630)